src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -139,14 +140,33 @@ def __init__(self, *args, **kwargs):
         super(TestWithServers, self).__init__(*args, **kwargs)
 
         self.agent_sessions = None
+        self.nvme_parameter = None
 
-    def setUp(self):
-        super(TestWithServers, self).setUp()
-
+    def init_server_param(self):
+        '''Read the server related parameter from avocado yaml file'''
+        self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
         self.server_group = self.params.get("server_group", '/server/',
                                             'daos_server')
-        self.hostlist_servers = self.params.get("test_machines", '/run/hosts/*')
 
+    def spdk_setup(self):
+        '''SPDK setup for NVMe drives'''
+        self.init_server_param()
+        self.nvme_parameter = self.params.get("bdev_class", '/server_config/')
```

Comment:
Will all tests that want to use NVMe set `bdev_class: nvme` in their `.yaml` file, triggering this `spdk.nvme_setup()` and `spdk.nvme_cleanup()` for each test `setUp()` and `tearDown()`?

Reply:
Yes for this item, Later this needs to be done based on Tags. In the past we were passing --nvme option to launch script which triggers the nvme setup and cleanup. There is another story to identify  the existing tests which needs to run with NVMe.